site/src/components/WorkspaceScheduleButton/WorkspaceScheduleButton.tsx

```diff
@@ -0,0 +1,249 @@
+import Button from "@material-ui/core/Button"
+import IconButton from "@material-ui/core/IconButton"
+import Popover from "@material-ui/core/Popover"
+import { makeStyles } from "@material-ui/core/styles"
+import Tooltip from "@material-ui/core/Tooltip"
+import AddIcon from "@material-ui/icons/Add"
+import RemoveIcon from "@material-ui/icons/Remove"
+import ScheduleIcon from "@material-ui/icons/Schedule"
+import cronstrue from "cronstrue"
+import dayjs from "dayjs"
+import advancedFormat from "dayjs/plugin/advancedFormat"
+import duration from "dayjs/plugin/duration"
+import relativeTime from "dayjs/plugin/relativeTime"
+import timezone from "dayjs/plugin/timezone"
+import utc from "dayjs/plugin/utc"
+import { useRef, useState } from "react"
+import { Workspace } from "../../api/typesGenerated"
+import { extractTimezone, stripTimezone } from "../../util/schedule"
+import { isWorkspaceOn } from "../../util/workspace"
+import { Stack } from "../Stack/Stack"
+import { WorkspaceSchedule } from "../WorkspaceSchedule/WorkspaceSchedule"
+
+// REMARK: some plugins depend on utc, so it's listed first. Otherwise they're
+//         sorted alphabetically.
+dayjs.extend(utc)
+dayjs.extend(advancedFormat)
+dayjs.extend(duration)
+dayjs.extend(relativeTime)
+dayjs.extend(timezone)
+
+export const Language = {
+  autoStartDisplay: (schedule: string | undefined): string => {
+    if (schedule) {
+      return (
+        cronstrue
+          .toString(stripTimezone(schedule), { throwExceptionOnParseError: false })
+          // We don't want to keep the At because it is on the label
+          .replace("At", "")
+      )
+    } else {
+      return "Manual"
+    }
+  },
+  autoStartLabel: "Starts at",
+  autoStopLabel: "Stops at",
+  workspaceShuttingDownLabel: "Workspace is shutting down",
+  autoStopDisplay: (workspace: Workspace): string => {
```

Comment:
Yes, I agree this one should live outside of the language object. About `useMemo`, even if we calculate this on every render the impact should be minimal. I like to delay optimizations as possible so we avoid adding "early complexity/layers".

Reply:
I agree with the avoiding early optimization (especially with some of the util functions). I think we are in an area where we may want to tread a little more carefully because we are polling for workspace every second or so. Am I missing something with XState? I'm concerned this view is going to be updated at that same cadence, which seems a bit overkill.

Reply:
Oof, I think we should do it!